### PR TITLE
Do not overwrite mbed_app.json if it exists

### LINKED ISSUE #848: 'mbed new' overwrites existing mbed_app.json
running mbed new command to existing codebase which includes mbed_app.json will overwrite the contents of mbed_app.json

### PR DESCRIPTION
**Pull request by bridadan**
Fixes #848.

### CODE REVIEW

File: mbed/mbed.py [hunk 1]
```diff
@@ -226,10 +228,10 @@ def hide_progress(max_width=80):
...
 def create_default_mbed_app():
     # Default data content
-    data = {'target_overrides':{'K64F':{'platform.stdio-baud-rate': 9600}}}
-    filehandler = open("mbed_app.json","w")
-    json.dump(data, filehandler, indent=4)
-    filehandler.close()
+    if not os.path.exists(mbed_app_file_name):
+        data = {'target_overrides':{'K64F':{'platform.stdio-baud-rate': 9600}}}
+        with open(mbed_app_file_name, "w") as mbed_app_file:
+            mbed_app_file(json.dumps(data, filehandler, indent=4))
```
[the remark this answers is not in the document]
Comment (bridadan) [reply mbed/mbed.py]: Oh man yeah, I goofed this whole thing up lol, thanks for the catch. I was trying to go too quick.

**Comment by bridadan** [comment]
Ok updated. I moved it to the `with` syntax just to avoid needing to explicitly close the file descriptor. Its a personal preference of mine but I can switch it back if you like.

**Comment by bridadan** [comment]
I think there's an issue with Mbed OS 5.11.4 and Python 3: https://github.com/ARMmbed/mbed-os/pull/9521#issuecomment-463848452

So the error here is not related to the change.

**Comment by cmonr** [comment]
@bridadan In the Python 3 PR that you commented in (not sure why it doesn't show up as a link here...), I commented that it looks like an error that was caught with a PR coming into 5.11.5.

Also mentioned that I'm looking to add basic compile tests in Travis CI to make sure this doesn't happen again... Not sure what you want to do from here.

**Pull request merged**